version: '3'


services: 
  keycloak:
    image: quay.io/keycloak/keycloak:20.0.1
    deploy:
      mode: replicated
      replicas: 1
      labels:
        - "traefik.enable=true"
        - "traefik.docker.network=traefik_public"
        - "traefik.http.routers.keycloak.entrypoints=websecure"
        - "traefik.http.routers.keycloak.rule=Host(`auth.localdomain`)"
        - "traefik.http.routers.keycloak.service=keycloak_service"
        - "traefik.http.routers.keycloak.tls=true"
        - "traefik.http.routers.keycloak.middlewares=keycloak_proto"
        - "traefik.http.services.keycloak_service.loadbalancer.server.port=8080"
        - "traefik.http.middlewares.keycloak_proto.headers.customrequestheaders.X-Forwarded-Proto=https"
    restart: on-failure
    environment:
      - KEYCLOAK_ADMIN=admin
      - KEYCLOAK_ADMIN_PASSWORD=admin
      - KC_PROXY=edge
      - KC_DB=postgres
      - KC_DB_URL=jdbc:postgresql://db/keycloak
      - KC_DB_PASSWORD=keycloak
      - KC_DB_USERNAME=keycloak
    command: start-dev --proxy=edge --hostname-url=https://auth.localdomain
    networks:
       overlay_swarm_net:
          ipv4_address: 192.168.60.5
       traefik_public:
    depends_on:
      - db
    


networks:
  overlay_swarm_net:
    name: overlay_swarm_net
    external: true

  traefik_public:
    driver: overlay
    name: traefik_public
    